File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        next = 0
        curr = 0
        ans = 0

        for i in range(0,n-1):
            next = max(next , i + nums[i])

            if curr == i:
                ans += 1
                curr = next

        return ans

File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        int ans = 0;
        int next = 0;
        int curr = 0;

        for(int i=0;i<n-1;i++){
            next = max(next , i + nums[i]);

            if(i == curr){
                ans++;
                curr = next;
            }
        }
        return ans;
    }
};